src/node/chainstatemanager_args.cpp

```diff
@@ -39,7 +39,14 @@ util::Result<void> ApplyArgsManOptions(const ArgsManager& args, ChainstateManage
         }
     }
 
-    if (auto value{args.GetArg("-assumevalid")}) opts.assumed_valid_block = uint256S(*value);
+    if (auto value{args.GetArg("-assumevalid")}) {
+        if (*value == "0") value.emplace(uint256::size() * 2, '0'); // handle -noassumevalid
```

Comment:
nit: if you think it's more readable, here's an alternative which avoids reparsing and duplicating `uint256::size() * 2` (but assigns twice, so you may not like it):
```C++
    if (auto value{args.GetArg("-assumevalid")}) {
        if (*value == "0") { // handle -noassumevalid
            opts.assumed_valid_block = uint256{};
        } else if (auto block_hash{uint256::FromHex(*value)}) {
            opts.assumed_valid_block = *block_hash;
        } else {
            return util::Error{strprintf(Untranslated("Invalid assumevalid block hash specified (%s), must be %d character hex (or 0 to disable)"), *value, uint256::size() * 2)};
        }
    }
```